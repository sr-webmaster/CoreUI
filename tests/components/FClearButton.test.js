import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/FClearButton.vue'

describe('FClearButton', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('clickButton() emits clear', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.clickButton()
      expect(wrapper.emitted().clear).toBeTruthy()
    })
  })
})
