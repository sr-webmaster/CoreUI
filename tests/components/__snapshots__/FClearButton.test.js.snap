// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FClearButton Snapshots defaults 1`] = `
<button
  class="mr-0 v-btn v-btn--flat v-btn--icon theme--light"
  type="button"
>
  <div
    class="v-btn__content"
  >
    <i
      aria-hidden="true"
      class="v-icon clear-button-icon fa fa fa fa-times theme--light"
    />
  </div>
</button>
`;
